### ISSUE TITLE
FormatKotlin task doesn't seem to format code in /kotlin sourceset.

### ISSUE THREAD

**Issue by paulwang007** [jeremymailen/kotlinter-gradle#115]
FormatKotlin task doesn't seem to work for code inside /src/main/kotlin/* sourcesets. But it works just fine for /src/main/java/*.

**Comment by jeremymailen** [comment]
Thanks for the report. Which kotlin plugin are you using -- jvm or android? Any customizations in the gradle build DSL I should note to reproduce it?

**Comment by paulwang007** [comment]
Plugins:
apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'
apply plugin: 'kotlin-android-extensions'

android {
  compileSdkVersion 29

  defaultConfig {
    minSdkVersion 21
    targetSdkVersion 29
    versionCode versionCode
    versionName version

    testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
  }

  buildTypes {
    release {
      useProguard false
    }
    debug {
      useProguard false
    }
  }

  compileOptions {
    // Source code java level(Source code of this project).
    sourceCompatibility '1.8'
    // Target JVM level(JVM on Android devices).
    targetCompatibility '1.8'
  }

  kotlinOptions {
    jvmTarget = '1.8'
  }

  sourceSets {
    main.java.srcDirs += 'src/main/kotlin'
    test.java.srcDirs += 'src/test/kotlin'
  }

  testOptions {
    unitTests.returnDefaultValues = true
    unitTests.includeAndroidResources = true
  }

  buildToolsVersion = '29.0.2'
}

**Comment by jeremymailen** [comment]
I think I see what's happening. kotlinter-gradle is resolving the `srcDirs` and assigning them to each lint and format task before the customizations are evaluated.

I think it's fixable if we use a gradle `Provider` or `Closure` to provide the final `srcDirs` as use time instead of configuration. I'll try to get something out this week.

**Comment by jonasbark** [comment]
The tasks don't even appear for Kotlin Multiplatform projects - is it possible that that's the same issue?

**Comment by jeremymailen** [comment]
I don't believe so @jonasbark, kotlinter doesn't yet support automatically creating tasks for the multiplatform plugin, although you could create them manually.

**Comment by mateuszkwiecinski** [comment]
I know the issue is quite outdated, but I faced exactly the same issue as the original one. 

From my understaniing calling: 
```
sourceSets {
    main.java.srcDirs += 'src/main/kotlin'
    test.java.srcDirs += 'src/test/kotlin'
}
```
could be easily replaced with:
```
android {
    sourceSets.whenObjectAdded { set ->
        def withKotlin = set.java.srcDirs.collect { it.path.replace("java", "kotlin") }
        set.java.setSrcDirs(set.java.srcDirs + withKotlin)
    }
}
```
(it also works for build variant specific sources like i.e. `src/debug/kotlin` or `src/release/kotlin`)
which happens early enough that kotlinter sets proper sources for _lint_ and _format_ tasks.

**Comment by jeremymailen** [comment]
Thanks for the tip @mateuszkwiecinski!

This issue is still very much alive and on my mind. I just need to find some time to refactor so that sources are lazily evaluated. PRs are also very welcome.

**Comment by jeremymailen** [comment]
Fixed in 2.3.0
